libs/server-sdk/src/client_impl.cpp

```diff
@@ -234,193 +240,239 @@ void ClientImpl::FlushAsync() {
     event_processor_->FlushAsync();
 }
 
-template <typename T>
```

Comment:
This has been totally refactored to more closely match our specification. There's now a Pre and Post step, which take care of all the details including sending events.